src/install/semver.zig

```diff
@@ -1418,113 +1439,125 @@ pub const Query = struct {
             switch (this.tag) {
                 // Allows changes that do not modify the left-most non-zero element in the [major, minor, patch] tuple
                 .caret => {
-                    var right_version = version;
-                    // https://github.com/npm/node-semver/blob/cb1ca1d5480a6c07c12ac31ba5f2071ed530c4ed/classes/range.js#L310-L336
-                    if (right_version.major == 0) {
-                        if (right_version.minor == 0) {
-                            right_version.patch += 1;
-                        } else {
-                            right_version.minor += 1;
-                            right_version.patch = 0;
-                        }
-                    } else {
-                        right_version.major += 1;
-                        right_version.patch = 0;
-                        right_version.minor = 0;
-                    }
-
-                    return Range{
-                        .left = .{
+                    // https://github.com/npm/node-semver/blob/3a8a4309ae986c1967b3073ba88c9e69433d44cb/classes/range.js#L302-L353
+                    var range = Range{};
+                    if (version.hasMajor()) done: {
+                        range.left = .{
                             .op = .gte,
-                            .version = version,
-                        },
-                        .right = .{
+                            .version = .{
+                                .major = version.major,
+                            },
+                        };
+                        range.right = .{
                             .op = .lt,
-                            .version = right_version,
-                        },
-                    };
+                        };
+                        if (version.hasMinor()) {
+                            range.left.version.minor = version.minor;
+                            if (version.hasPatch()) {
+                                range.left.version.patch = version.patch;
+                                if (version.major == 0) {
+                                    if (version.minor == 0) {
+                                        range.right.version.patch = version.patch + 1;
```

Comment:
```suggestion
                                        range.right.version.patch = version..patch +| 1;
```

Reply:
is the extra `.` intentional?

Reply:
Done in https://github.com/oven-sh/bun/pull/1854/commits/f0e686aa6b42639a78f758b9a9c76b6d046d1d85